### PR TITLE
WIP: cleanup 2017 branch for merge into master

### PR DESCRIPTION
**Pull request by mpacer**
This isn't ready yet, but I've made some progress using `git filter-branch` to get a cleaner branch to merge into master and then to base a new `dev` branch off of.

I should have this ready by the end of the day.

### CODE REVIEW

**Comment by mpacer** [comment]
So there are still some empty merge commits that I can't figure out why they aren't being removed from my `git filter-branch` call. 

For posterity, I think it's worth noting how this was done, it required a few steps.

The first few were fairly straightforward.

1. we need to remove all the paper directories from the index history:

    ```
    git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch -r papers/brendt_wohlberg papers/geoffrey_poore papers/ingo_heimbach papers/joseph_hellserstein papers/lindsay papers/manuel_krebber papers/michael_lange papers/oleksandr_pavlyk papers/will_barnes papers/dillon_niederhut papers/horea-ioan_ioanas papers/jaime_arias papers/klaus_greff papers/mahzad_khoslessan papers/michael_beyeler papers/narendra_mukherjee papers/scott_sievert papers/joseph_hellerstein' 070cd84a022305b..HEAD
    ```

2. This leaves a bunch of empty commits — let's get rid of those:

    ```
    git filter-branch -f --prune-empty 070cd84a022305b..HEAD
    ```

3. Then I realised that some people had commits that weren't supposed to ever have existed, specifically into the `output/` directory so the solution is:

    ```
    git filter-branch -f --index-filter 'git rm --cached --ignore-unmatch -r output/' 070cd84a022305b..HEAD
    ```

4. Which left more empty commits…

    ```
    git filter-branch -f --prune-empty 070cd84a022305b..HEAD
    ```

5. Alright, so we got all of the content… but this leaves a bunch of empty merge commits… after a bit of searching I found http://jimmy.schementi.com/splitting-up-a-git-repo. Specifically under the section titled `Arbitrary Paths`:

    ```
    tmpfile=`mktemp -t splitgit`
    chmod 755 $tmpfile
    echo '#!/usr/bin/env ruby
    old_parents = gets.chomp.gsub("-p ", " ")
    new_parents = old_parents.empty? ? [] : `git show-branch --independent #{old_parents}`.split
    puts new_parents.map{|p| "-p " + p}.join(" ")' > $tmpfile
    git filter-branch -f --prune-empty --parent-filter $tmpfile 070cd84a022305b..HEAD
    rm $tmpfile
    ```

6. This left more empty commits around so another `--prune-empty`…

    ```
    git filter-branch -f --prune-empty 070cd84a022305b..HEAD
    ```

And we're at the place where we are now. 

There are still a few renegade merge commits that I wasn't able to fix with this approach 
(325fd5c, 7e5fe76). But I think this is good enough to suffice cleaning up the 2017 branch to merge into master.

**Comment by stefanv** [comment]
Awesome, great work @mpacer !  This satisfies my preference that the binary paper data not be included along with the tools (not even in history), so I am +1.

**Comment by mpacer** [comment]
@dalippa @deniederhut @FatihAkici if any of you have comments please say something before this evening I’ll be merging this and then creating the new 2018 and dev branches this evening.

File: README.md [hunk 1]
```diff
@@ -102,38 +104,39 @@ examples of how to:
...
 ## Build Server
 
-Thanks to the great and wonderful Katy Huff, there is a server online 
-building the open pull requests [here](http://zibi.bids.berkeley.edu:7001). You may be 
-able to pull a built PDF for review from there.
+There will be a server online building the open pull requests 
+[here](http://zibi.bids.berkeley.edu:7001). You should be able to pull a built PDF 
+for review from there.
+
+TODO: update server link
```
Comment (dalippa) [inline README.md]: Are you planning on updating this to the new build server?

**Review by dalippa — APPROVED** [review]
One comment @mpacer but it needn't be fixed tonight before the merge, but it does need to be fixed before submitters start posting papers and looking at the build server.

File: README.md (hunk 1 above)
Comment (mpacer) [reply README.md]: Yes, but this was an attempt to change literally nothing but to bring master up-to-date with 2017 minus the paper commits.

**Comment by mpacer** [comment]
Ok going to merge.

**Comment by mpacer** [comment]
Lol. should have changed the title before merging, but I wasn't thinking. Oh well!

**Pull request merged**